0001-kaa-protocol/README.md

```diff
@@ -7,83 +7,99 @@ contributors: Alexey Gamov <agamov@cybervisiontech.com>
+There are two large classes of extensions:
+- _endpoint-aware extensions_ operate against a defined endpoint.
+For example, data collection extension requires the source endpoint identification.
 
-_Note: we can assign endpoint tokens per client session. That’s more secure and allows using short non-secure tokens._
+- _endpoint-unaware extensions_ do not require the endpoint identity to function.
+For example, endpoint registration extension may function before there is an established endpoint identity in the server.
 
-On the other hand, using a predefined token is a simpler but less secure alternative. In that case, the endpoint is manually added to the server, the server allocates a unique token and provides it to the user, the user embeds the token in the application.
+The same extension may be configured in a Kaa server more than once.
+For example, there may be several instances of the data collection extension set up for collecting data of different nature.
+_Extension instance names_ are used in Kaa to distinguish between _extension instances_.
 
 ### Resource path format
+Both MQTT and CoAP support some kind of a _resource path_: MQTT has hierarchical topic names, and CoAP has URI-Path.
+In KP they are used to differentiate types of requests and responses.
 
-Both MQTT and CoAP have some kind of _resource path_: MQTT has hierarchical topic names, and CoAP has URI-Path. It is wise to use them to differentiate requests.
-
-We divide resource path into two parts. The first part of the path identifies a specific extension instance, and the second part is extension-specific. That allows extensions to have their own resource hierarchies. This document only describes the first part and gives general recommendation for the second one.
-
-Note that MQTT is sensitive to trailing/leading slashes: **a/b/c** is not same as **a/b/c/** or **/a/b/c**.
+Resource path is separated into two parts.
+The first part is common for all extensions, and the second is extension-specific.
+That allows extensions define their own resource path hierarchies.
+This document only describes the first part and provides requirements and recommendations for the second one.
 
-The common path has the next form:
-```
-kaa/<application_token>/<extension_instance_id>
-```
+**Note:** MQTT is sensitive to trailing/leading slashes: `a/b/c` is not the same as `a/b/c/` or `/a/b/c`.
 
-All paths have a `kaa` prefix. (It’s likely to be configurable in the future.) The prefix helps routing all KP-related traffic though MQTT brokers, as all data can be matched with a single topic filter: `kaa/#`.
+The common resource path part has the following format:
+`kaa/<appversion_name>/<extension_instance_name>`
 
-`<application_token>` is a unique token that identifies application and its version within a server instance.
+All KP resource paths start with `kaa`, which is the reserved prefix for KP version 1.
+Future versions of KP will have prefixes such as `kaav2`, `kaav3`, and so on.
```

Comment:
"are expected to have prefixes" 

BTW, it might be better to use "kp" instead of "kaa"?